R/homelessness_flag.R

```diff
@@ -0,0 +1,146 @@
+#'
+#' For data since 2018, previous studies have
+#' shown that "ICD-10-CA codes have a sensitivity of 60-70% for
+#' detecting whether a patient has experienced homelessness at the
+#' time of hospitalization" (Richard et al, 2024). Due to this,
+#' we recommend to only use ICD-10-CA codes to identify
+#' homelessness after 2018.
+#'
+#' @param cohort (`data.frame` or `data.table`)
+#' Cohort table with all relevant encounters of interest, where each row
+#' corresponds to a single encounter. Must contain GEMINI Encounter ID
+#' (`genc_id`).
+#'
+#' @param ipdiag (`data.table`)
+#' `ipdiagnosis` table as defined in the [GEMINI Data Repository Dictionary](https://drive.google.com/uc?export=download&id=1iwrTz1YVz4GBPtaaS9tJtU0E9Bx1QSM5).
+#' This table must contain `genc_id` and `diagnosis_code` (as ICD-10-CA
+#' alphanumeric code) in long format.
+#'
+#' @param erdiag (`data.table`)
+#' `erdiagnosis` table as defined in the [GEMINI Data Repository Dictionary](https://drive.google.com/uc?export=download&id=1iwrTz1YVz4GBPtaaS9tJtU0E9Bx1QSM5).
+#' This table must contain `genc_id` and `er_diagnosis_code` (as ICD-10-CA
+#' alphanumeric code) in long format.
+#'
+#' Typically, ER diagnoses should be included when deriving the
+#' homelessness flags in order to increase sensitivity. However, in
+#' certain scenarios, users may choose to only include IP diagnoses by
+#' specifying `erdiag = NULL`. This may be useful when comparing cohorts
+#' with different rates of ER admissions.
+#'
+#' @return
+#' data.table with the same number of rows as input `cohort` with an
+#' additional derived boolean field labelled as `"homelessness_icd__flag"`.
+#' Possible values are `TRUE`, `FALSE` or `NA`. `NA` indicates that an
+#' encounter does not have a diagnosis code in the diagnosis table input.
+#'
+#' @examples
+#' \dontrun{
+#' drv <- dbDriver("PostgreSQL")
+#' dbcon <- DBI::dbConnect(drv,
+#'   dbname = "report_db_v3",
+#'   host = "prime.smh.gemini-hpc.ca",
+#'   port = 5432,
+#'   user = getPass("Enter user:"),
+#'   password = getPass("password")
+#' )
+#' ipadm <- dbGetQuery(dbcon, "select * from admdad") %>% data.table()
+#' ipdiagnosis <- dbGetQuery(dbcon, "select * from ipdiagnosis") %>% data.table()
+#' erdiagnosis <- dbGetQuery(dbcon, "select * from erdiagnosis") %>% data.table()
+#' homeless <- homelessness_flag(cohort = ipadm, ipdiag = ipdiagnosis)
+#' # view only genc_id's with homelessness flag
+#' homeless <- homeless %>% filter(homeless$homelessness_icd_flag == TRUE)
+#' }
+#'
+#' @references
+#' \itemize{
+#'    \item{[ALC Diagnosis List](https://www.cihi.ca/sites/default/files/document/alternate-level-care-diagnosis-list-job-aid-en.pdf)}
+#'    \item{Identification of homelessness using health administrative data in Ontario: Richard Lucie, et al. J. Clin. Epidimiol., 2024. https://doi.org/10.1016/j.jclinepi.2024.111430}
+#' }
+#'@export
+homelessness_flag <- function(
+    cohort,
+    ipdiag,
+    erdiag = NULL) {
```

Comment:
You actually don't want to specify `erdiag = NULL` here. The idea is that users are "forced" to specify `erdiag = NULL` if they indeed want to exclude ER diagnosis codes. That way they can't accidentally omit ER diagnoses without realizing it. Note that this actually also causes one of the unit tests to fail (see below). 
```suggestion
    erdiag) {
```